pkg/workers/leader_election_mgr_test.go

```diff
@@ -185,3 +187,465 @@ func TestLeaderElectionManager_acquireLeaderLease(t *testing.T) {
+	type fields struct {
+		workers                                []Worker
+		connectionFactory                      *db.ConnectionFactory
+		tearDown                               chan struct{}
+		leaderElectionReconcilerRepeatInterval time.Duration
+		workerGrp                              *sync.WaitGroup
+	}
+	var wg sync.WaitGroup
+	wg.Add(1)
+	tests := []struct {
+		name    string
+		fields  fields
+		setupFn func()
+	}{
+		{
+			name: "should return if teardown is nil",
+			fields: fields{
+				tearDown: nil,
+				workers: []Worker{
+					&WorkerMock{
+						GetIDFunc: func() string {
+							return "02"
+						},
+						GetWorkerTypeFunc: func() string {
+							return "leader"
+						},
+						IsRunningFunc: func() bool {
+							return true
+						},
+						GetSyncGroupFunc: func() *sync.WaitGroup {
+							return &sync.WaitGroup{}
+						},
+						StopFunc: func() {
+						},
+					},
+				},
+				connectionFactory:                      nil,
+				leaderElectionReconcilerRepeatInterval: 15 * time.Second,
+				workerGrp:                              &sync.WaitGroup{},
+			},
+			setupFn: nil,
+		},
+		{
+			name: "should stop the worker",
+			fields: fields{
+				tearDown:  make(chan struct{}),
+				workerGrp: &wg,
+				workers: []Worker{
+					&WorkerMock{
+						GetIDFunc: func() string {
+							return "02"
+						},
+						GetWorkerTypeFunc: func() string {
+							return "leader"
+						},
+						IsRunningFunc: func() bool {
+							return true
+						},
+						GetSyncGroupFunc: func() *sync.WaitGroup {
+							return &wg
+						},
+						StopFunc: func() {
+							wg.Done()
+						},
+					},
+				},
+				leaderElectionReconcilerRepeatInterval: 15 * time.Second,
+				connectionFactory:                      db.NewMockConnectionFactory(nil),
+			},
+			setupFn: func() {
+				mockEntry := map[string]interface{}{
+					"leader":  "02",
+					"expires": time.Now().Add(time.Hour),
+				}
+				mocket.Catcher.Reset().
+					NewMock().
+					WithQuery("SELECT * FROM leader_leases").
+					WithReply([]map[string]interface{}{mockEntry})
+				mocket.Catcher.NewMock().WithExecException().WithQueryException()
+			},
+		},
+	}
+	for _, testcase := range tests {
+		tt := testcase
+		t.Run(tt.name, func(t *testing.T) {
+			g := gomega.NewWithT(t)
+			t.Parallel()
+			if tt.setupFn != nil {
+				tt.setupFn()
+			}
+			s := &LeaderElectionManager{
+				workers:                                tt.fields.workers,
+				connectionFactory:                      tt.fields.connectionFactory,
+				tearDown:                               tt.fields.tearDown,
+				leaderElectionReconcilerRepeatInterval: tt.fields.leaderElectionReconcilerRepeatInterval,
+				workerGrp:                              *tt.fields.workers[0].GetSyncGroup(),
+			}
+			if tt.fields.tearDown != nil {
+				s.Start()
+				s.Stop()
+				workerMock := tt.fields.workers[0].(*WorkerMock)
+				g.Expect(workerMock.calls.Stop).To(gomega.HaveLen(1))
+			}
+			s.Stop()
```

Comment:
If `tearDown` is nil, no tests are performed, making the first test ("should return if teardown is nil") meaningless